backend/src/test/java/com/bikes/backend/service/BikeServiceTest.java

```diff
@@ -3,43 +3,120 @@
 import com.bikes.backend.model.Bike;
 import com.bikes.backend.repository.BikeRepository;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
-import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
+import java.util.Optional;
 
-import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.*;
+
 class BikeServiceTest {
 
-    BikeRepository bikeRepository = mock(BikeRepository.class);
-    BikeService bikeService = new BikeService(bikeRepository);
+	BikeRepository bikeRepository = mock(BikeRepository.class);
+	IdService idService = mock(IdService.class);
+    BikeService bikeService = new BikeService(bikeRepository, idService);
 
-    Bike testBike = new Bike("testBike", "testId");
-    List<Bike> expectedBikes = List.of(testBike);
-    List<Bike> expectedBikesEmpty = new ArrayList<>();
+	Bike testBike = new Bike("testId", "testBike");
+	List<Bike> expectedBikes = List.of(testBike);
+	List<Bike> expectedBikesEmpty = new ArrayList<>();
 
-    @Test
-    void getAllBikesNonEmptyRepo() {
-        //GIVEN
-        when(bikeRepository.getAllBikes()).thenReturn(expectedBikes);
-        //WHEN
-        List<Bike> actual = bikeService.getAllBikes();
-        //THEN
-        verify(bikeRepository).getAllBikes();
-        Assertions.assertEquals(expectedBikes, actual);
-    }
 
+	@Nested
+	@DisplayName("testing getAllBikes()")
+	class getAllBikesTest {
+		@Test
+		@DisplayName("...returns all bikes if the repo is not empty")
+		void getAllBikes_returnsAllBikesIfTheRepoIsNotEmpty() {
+			//GIVEN
+			when(bikeRepository.getAllBikes()).thenReturn(expectedBikes);
+			//WHEN
+			List<Bike> actual = bikeService.getAllBikes();
+			//THEN
+			verify(bikeRepository).getAllBikes();
+			Assertions.assertEquals(expectedBikes, actual);
```

Comment:
expectedBikes will be better placed before verify, so it is easier to read

Reply:
No